api/soap/mc_issue_api.php

```diff
@@ -436,8 +436,7 @@ function mci_issue_set_monitors( $p_issue_id, $p_requesting_user_id, array $p_mo
 	# 2. build new monitors ids
 	$t_new_monitor_ids = array();
 	foreach ( $p_monitors as $t_monitor ) {
-		$t_monitor = SoapObjectsFactory::unwrapObject( $t_monitor );
-		$t_new_monitor_ids[] = $t_monitor['id'];
+		$t_new_monitor_ids[] = mci_get_user_id( $t_monitor );
```

Comment:
email_ensure_unique has been introduced in 1.3.
You might have non unique email address after upgrading from older versions although email_ensure_unique == ON